staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -586,25 +668,30 @@ func (e *Store) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 
 	if creating {
 		if e.AfterCreate != nil {
-			if err := e.AfterCreate(out); err != nil {
-				return nil, false, err
-			}
+			e.AfterCreate(out)
 		}
 	} else {
 		if e.AfterUpdate != nil {
-			if err := e.AfterUpdate(out); err != nil {
-				return nil, false, err
-			}
+			e.AfterUpdate(out)
 		}
 	}
 	if e.Decorator != nil {
-		if err := e.Decorator(out); err != nil {
-			return nil, false, err
-		}
+		e.Decorator(out)
 	}
 	return out, creating, nil
 }
 
+// This is a helper to convert UpdateOptions to CreateOptions for the
+// create-on-update path.
+func newCreateOptionsFromUpdateOptions(in *metav1.UpdateOptions) *metav1.CreateOptions {
```

Comment:
Can we get a test that ensures we're handling all fields? Something like we did for the `rest.CopyConfig`, that makes us look at this method every time we add a value to UpdateOptions or to CreateOptions.

Reply:
context https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/config_test.go#L395-L493

Reply:
ACK.. It's a little gross, but it works.